src/shared/particle_dynamics/continuum_dynamics/continuum_integration.cpp

```diff
@@ -116,14 +135,116 @@ void StressDiffusion::interaction(size_t index_i, Real dt)
     }
     stress_rate_3D_[index_i] = diffusion_stress_rate_;
 }
 //====================================================================================//
+ShearStressRelaxationHourglassControl ::
+    ShearStressRelaxationHourglassControl(BaseInnerRelation &inner_relation, Real xi)
+    : fluid_dynamics::BaseIntegration<DataDelegateInner>(inner_relation),
+      continuum_(DynamicCast<GeneralContinuum>(this, particles_->getBaseMaterial())),
+      shear_stress_(particles_->registerStateVariable<Matd>("ShearStress")),
+      shear_stress_rate_(particles_->registerStateVariable<Matd>("ShearStressRate")),
+      velocity_gradient_(particles_->registerStateVariable<Matd>("VelocityGradient")),
+      strain_tensor_(particles_->registerStateVariable<Matd>("StrainTensor")),
+      strain_tensor_rate_(particles_->registerStateVariable<Matd>("StrainTensorRate")),
+      B_(particles_->getVariableDataByName<Matd>("LinearGradientCorrectionMatrix")),
+      von_mises_stress_(particles_->registerStateVariable<Real>("VonMisesStress")),
+      von_mises_strain_(particles_->registerStateVariable<Real>("VonMisesStrain")),
+      scale_penalty_force_(particles_->registerStateVariable<Real>("ScalePenaltyForce", Real(1.0))),
+      acc_shear_(particles_->registerStateVariable<Vecd>("AccelerationByShear")),
+      acc_hourglass_(particles_->registerStateVariable<Vecd>("AccelerationHourglass")),
+      G_(continuum_.getShearModulus(continuum_.getYoungsModulus(), continuum_.getPoissonRatio())), xi_(xi)
+{
+    particles_->addVariableToSort<Matd>("ShearStress");
+    particles_->addVariableToSort<Matd>("ShearStressRate");
+    particles_->addVariableToSort<Matd>("VelocityGradient");
+    particles_->addVariableToSort<Matd>("StrainTensor");
+    particles_->addVariableToSort<Matd>("StrainTensorRate");
+    particles_->addVariableToSort<Real>("VonMisesStress");
+    particles_->addVariableToSort<Real>("VonMisesStrain");
+    particles_->addVariableToSort<Real>("ScalePenaltyForce");
+    particles_->addVariableToSort<Vecd>("AccelerationByShear");
+    particles_->addVariableToSort<Vecd>("AccelerationHourglass");
+}
+//====================================================================================//
+void ShearStressRelaxationHourglassControl::initialization(size_t index_i, Real dt)
+{
+    Matd velocity_gradient = Matd::Zero();
+    Neighborhood &inner_neighborhood = inner_configuration_[index_i];
+    Vecd vel_i = vel_[index_i];
+    for (size_t n = 0; n != inner_neighborhood.current_size_; ++n)
+    {
+        size_t index_j = inner_neighborhood.j_[n];
+        Real dW_ijV_j = inner_neighborhood.dW_ij_[n] * Vol_[index_j];
+        Vecd &e_ij = inner_neighborhood.e_ij_[n];
+        Matd velocity_gradient_ij;
+        velocity_gradient_ij = -(vel_i - vel_[index_j]) * (B_[index_i] * e_ij * dW_ijV_j).transpose();
+        velocity_gradient += velocity_gradient_ij;
+    }
+    velocity_gradient_[index_i] = velocity_gradient;
+}
```

Comment:
Could we call the inner_neighborhood to calculate gradient/divergence at 'initialization' and 'update'?

Reply:
Since I want to combine all the shear stress-related calculations in one Class, I also used the particle neighbors in `initialization`.
If this is not allowed, I will split the calculation of shear stress into two Classes.